feat: Add tutorials section (#93)
beclab/docs: docs/manual/tutorials/set-custom-domain.md

---
outline: [2, 3]
---

# Set up a custom domain for your Olares 

> This tutorial walks you through setting up your own domain for your Olares. By the end, you’ll have a custom domain bound to Olares ID and be able to install and activate Olares with it.

Reviewer: If this tutorial also applies to set up organiazational Olares ID, it should be clarified in the beginning too.

## Before you begin

Ensure you have:
- Your own domain registered through a domain registrar. 

  :::tip Note
  Only eTLD1 or eTLD2 domains are supported.
  :::
- LarePass app installed on your phone.

## Step 1: Create a DID

You can only bind a custom domain to the account when it is in the DID stage (not yet bound to an Olares ID). To create a DID:

![create DID](/images/manual/tutorials/create-did.png)
1. In LarePass app, go to the account creation page.
2. Tap **Create an account** to trigger a DID creation. 

This gets you an Olares account in the DID stage. 

## Step 2: Add your domain to Olares Space
Add and verify your own domain in Olares Space before binding it.

1. In your browser, access Olares Space at https://space.olares.xyz/.
2. In LarePass app, click the scan button in the top-right corner, and scan the QR code on the login page to log in to Olares Space.   
3. In Olares Space, go to **Domain management** > **Domain name setup**, enter your domain and click **Confirm**.
  ![add domain](/images/manual/tutorials/add-domain.png)
4. Verify your TXT record for your domain.
    <br>a. Click the **Guide** button next to the added domain. 

>     <br>b. Follow the on-screen instructions on the pop-up window to add a TXT record to your DNS provider configuration.

Reviewer: Use full name then abbreviation?

- Both Gmail and G-suite mail addresses are supported. Learn more in [Manage your domain](.././space/manage-domain.md#set-email-invitation-rules).
:::

## Step 5: Create an Olares ID for the member

To use the domain, create an Olares ID for the member by joining the created organization.

![Join the org](/images/manual/tutorials/join-org.png)

1. In the LarePass app, go to **Advanced Account Creation** > **Organization Olares ID** > **Join an existing organization**.
2. Type the org domain name (the verified custom domain) and click **Continue**. If you see an error, verify if the domain name is correct and the domain rules are set properly in Olares Space.
3. Add a VC for the member.
  <br>a. When prompted, select Google as your VC credential provider.
  <br>b. Log in with the Gmail account you added in the previous step and grant access for VC.  
 
 After successful authorization, your Olares ID with the custom domain, `justtest1953@xxxx.cloud`, is successfully created. The member status on the **Domain management** page becomes **Verified**.

## Step 6: Install & activate Olares
Almost there! Now you are all set to install and activate Olares using the customized Olares ID.

> 1. Update the installation script with the environment variables of your own domain and local name, and execute it to start the installation:
>     
>     ```bash
>     export TERMINUS_OS_DOMAINNAME=xxxx.cloud \
>       && export TERMINUS_OS_USERNAME=justtest1953 \ 
>       && curl -sSfL https://olares.sh | bash -
>     ```
> 

Reviewer: ```suggestion
1. Use the installation script with specified environment variables to start the installation:
    
    ```bash
    export TERMINUS_OS_DOMAINNAME=xxxx.cloud \
      && export TERMINUS_OS_USERNAME=justtest1953 \ 
      && curl -sSfL https://olares.sh | bash -
    ```
- `export TERMINUS_OS_DOMAINNAME=xxxx.cloud`: Specify your custom domain.
- `export TERMINUS_OS_USERNAME=justtest1953`: Specify the username (local name).
```

> 2. Wait for the installation to finish.  Depending on your network, the process can take 20–30 minutes. When the installation completes, you will see the prompt like below:

Reviewer: ```suggestion
2. Wait for the installation to finish.  Depending on your network, the process can take 20–30 minutes. When the installation completes, you will see the wizard URL and login credentials:
```

The table below outlines the steps involved in setting up a custom domain and who is responsible for each task. Depending on whether you're an individual user or part of an organization, the actions you need to perform will differ.

| Step                                        | Individual user | Organization admin | Organization member |
|---------------------------------------------|-----------------|--------------------|---------------------|
| Create a DID                                | ✅               | ✅                  | ✅                   |
| Add domain to Olares Space                  | ✅               | ✅                  |                     |
| Create organization for the domain          | ✅               | ✅                  |                     |
| Add email to the organization               | ✅               | ✅                  |                     |
| Join the organization & create an Olares ID | ✅               | ✅                  | ✅                   |

> | Set up Olares                               | ✅               | ✅                  | ✅                   |

Reviewer: To provide some contexts. But i think a flowchart will be better. Now let’s settle for table.
